appdx-bitcoindevkit.asciidoc

```diff
@@ -0,0 +1,709 @@
+
+The https://bitcoindevkit.org/:[Bitcoin Dev Kit] project is a set of libraries and tools used to create Bitcoin descriptor based wallets. The core https://github.com/bitcoindevkit/bdk:[`bdk`] library is written in Rust and can easily integrate Bitcoin wallet features into any application. BDK also provides language-bindings via the https://github.com/bitcoindevkit/bdk-ffi:[`bdk-ffi`] project that allow developers to create bitcoin wallets in their native language such as Kotlin/Java, Swift, and Python. BDK is designed to be modular and supports different blockchain clients for syncing wallet transactions, such as Electrum and custom databases for storing wallet data. This allows developers to choose the components that are right for their application.
+
+=== BDK Library Features
+
+Below is a summary of the BDK library features. All features and the most flexibility can be found by using the core Rust based library. A large subset of features are also provided to application developers via the Kotlin, Swift and Python language bindings. The Kotlin and Swift versions of BDK are also packaged for mobile developers to easily include in their Android or iOS projects.
+
+[frame=all, grid=all]
+[%header, cols="1,1,1,1,1"]
+|===
+| 
+|Rust
+|Kotlin
+|Swift
+|Python 
+
+|https://bitcoindevkit.org/descriptors/[Descriptor] defined spending conditions
+|✓
+|✓
+|✓
+|✓
+
+|Top level descriptor and address types +
+   * Legacy: PK, PKH, SH(WPKH), SH(WSH), SH +
+   * Segwitv0 (Bech32): WPKH, WSH +
+   * Segwitv1 (Bech32m, Taproot): TR +
+|✓ 
+|✓
+|✓
+|✓
+
+| Hierarchical Deterministic (HD) key paths
+|✓
+|✓
+|✓
+|✓
+
+| Networks: regtest, signet, testnet, bitcoin
+|✓
+|✓
+|✓
+|✓
+
+| Pluggable blockchain clients +
+  * Electrum, optional SOCKS5 proxy +
+  * Esplora, optional SOCKS5 proxy +
+  * Received TX verified against core consensus rules +
+  * SOCKS5 Proxy configuration +
+|✓
+|✓
+|✓
+|✓
+
+| Generate/restore master key mnemonic +
+  * 12-24 words with optional password +
+|✓
+|✓
+|✓
+|✓
+
+| Pluggable databases for wallet data +
+  * SQLite +
+  * Memory +
+|✓
+|✓
+|✓
+|✓
+
+| Wallet functions +
+  * Get new address +
+  * List Transactions +
+  * Get Balance +
+  * Create, sign, combine and Finalize PSBT +
+  * Branch and bound coin selection +
+  * Sync blockchain data +
+  * Manual coin control +
+  * Drain wallet +
+  * Replace by Fee (RBF) +
+  * Child pays for parent (CPFP) +
+  * Single recipient and amount +
+  * Multiple recipients and amounts (batch payment) +
+  * Add OP_RETURN output +
+  * Set TX fee rate or absolute fee +
+|✓
+|✓
+|✓
+|✓
+
+| Linux, MacOS Support
+|✓
+|✓
+|✓
+|✓
+
+| Android Support
+|
+|✓
+|
+|
+
+| iOS Support
+|
+|
+|✓
+|
+
+| Descriptor templates +
+  * BIP44 (PKH), BIP49 (SH_WPKH), BIP84 (WPKH) +
+|✓
+|
+|
+|
+
+| Pluggable blockchain clients +
+  * Compact block filters, single peer +
+  * Core RPC, cookie or passwd auth +
+|✓
+|
+|
+|
+
+| Pluggable coin selection +
+  * Custom user provided algorithm +
```

Comment:
We also have the branch and bound algorithm and the single random draw :)

Reply:
> We also have the branch and bound algorithm and the single random draw :)
We have branch and bound on line 80 under Wallet functions. I don't think we currently have a way to do random draw on it's own, only as a fallback if there's no good branch and bound solution right? But maybe with the 1.0 changes that will be an option we can add.

Reply:
Oh right, single random draw it's just a fallback. Yeah, this should change with bdk-core 1.0. But I really think it's important to add that we do have BnB, as the other two algorithms are really, really crappy privacy wise and fee wise...

Reply:
Apparently I was misreading the document, bnb is mentioned above, sorry :)